app/src/components/evals/EvalForm/EvalForm.tsx

```diff
@@ -0,0 +1,192 @@
+import { Input, VStack, Text, Checkbox, HStack } from "@chakra-ui/react";
+import { ExternalModel } from "@prisma/client";
+import { useMemo } from "react";
+import AutoResizeTextArea from "~/components/AutoResizeTextArea";
+import InfoCircle from "~/components/InfoCircle";
+import InputDropdown from "~/components/InputDropdown";
+import frontendModelProvider from "~/modelProviders/openai-ChatCompletion/frontend";
+import { getOutputTitle } from "~/server/utils/getOutputTitle";
+import { ORIGINAL_MODEL_ID } from "~/types/dbColumns.types";
+import { predefinedEvaluationModels } from "~/utils/externalModels/evaluationModels";
+import { useDataset, useNodeEntries } from "~/utils/hooks";
+
+interface EvalNameProps {
+  datasetId: string;
+  nameState: {
+    name: string;
+    setName: React.Dispatch<React.SetStateAction<string>>;
+  };
+  modelState: {
+    includedModelIds: string[];
+    setIncludedModelIds: React.Dispatch<React.SetStateAction<string[]>>;
+  };
+  rowsState: {
+    numRows: number;
+    setNumRows: React.Dispatch<React.SetStateAction<number>>;
+    numComparisons: number;
+  };
+  evaluationModelState: {
+    selectedEvaluationModel: ExternalModel;
+    setSelectedEvaluationModel: React.Dispatch<React.SetStateAction<ExternalModel>>;
+  };
+  instructionsState: {
+    instructions: string;
+    setInstructions: React.Dispatch<React.SetStateAction<string>>;
+  };
+}
+
+const EvalForm: React.FC<EvalNameProps> = ({
+  datasetId,
+  nameState,
+  modelState,
+  rowsState,
+  evaluationModelState,
+  instructionsState,
+}) => {
+  const { name, setName } = nameState;
+  const { includedModelIds, setIncludedModelIds } = modelState;
+  const { numRows, setNumRows, numComparisons } = rowsState;
+  const { selectedEvaluationModel, setSelectedEvaluationModel } = evaluationModelState;
+  const { instructions, setInstructions } = instructionsState;
+
+  const additionalInputPayloadTokens = 40;
```

Comment:
Good idea to account for additional tokens. How did you calculate this number?

Reply:
Also, I would uppercase these const variables and declare them above the function to increase readability and communicate that they aren’t derived from the inputs to this function.

Reply:
Kyle precalculated these numbers approximately.
I think I can also calculate the `additionalInputPayloadTokens` based on the `instructions` input. It may be relevant if clients give large instructions for evaluation.

Uppercase is a nice idea, thank you!